osquery/tables/system/windows/objects.cpp

```diff
@@ -0,0 +1,214 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <Windows.h>
+
+#include <osquery/core.h>
+#include <osquery/core/windows/kobjhandle.h>
+#include <osquery/core/windows/ntapi.h>
+#include <osquery/core/windows/wmi.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+// the windows object namespace consists of nested directories,
+// beginning with \
+//
+// each directory consists of zero or more entries, each of which consists
+// of an unqualified object name and an object type name.  both are
+// represented by the windows kernel as UTF-16 strings
+//
+// the pair of object name and object type name are used frequently
+// in this module.  use a std::pair of std::wstrings to represent this
+// pair and provide a simpler name
+//
+// this type definition is internal to this module and meant as a
+// type to be used with the rest of osquery.  this note is important
+// because osquery uses UTF-8 encoded std::strings throughout.  Because
+// windows uses UTF-16 strings, which are most easily represented by
+// std::wstring, std::wstrings are used internally to this module and
+// converted to UTF-8 encoded std::strings when communicating with the
+// rest of osquery
+//
+typedef std::pair<std::wstring, std::wstring> obj_name_type_pair;
+
+// helper to convert a std::wstring to an integer
+// std::stoi can throw, and std::stol doesn't support std::wstring
```

Comment:
@muffins pointed out that there is another helper already that accomplishes this.

Reply:
@muffins @zacbrown the helpers in conversions.h do NOT support std::wstring.  for reasons I documented in the code, I believe I'm compelled to support std::wstring. here's why:

to support object enumeration, I *need* to take output parameters in the form of strings _from_ windows apis and pass them _to_ other windows apis, again as strings, as input parameters.  because windows uses utf-16 strings internally, it is possible that object names can be outside any 8 bit character set and so therefore the only reasonable 8 bit encoding that will work is UTF-8.  That said, I'd be compelled to convert UTF-8 _to_ UTF-16, pass to windows, then convert output back to UTF-8 to use std::string consistently, then re-convert to UTF-16 prior to invoking next Windows API, convert output back to UTF-8, etc.  To net that all out, it seems to make sense to use std::wstring _internaly to table implementation module (objects.cpp)_ and convert to utf-8 (std::string) when returning back to the rest of osquery.

That's a long-winded way of saying that I believe we need std::swstring support in the helper functions.

note that I chose to maintain camel case in the helper in conversions.h, which is NOT consistent with the rest of conversions.h.  I'm happy to convert, just not sure how you guys want to proceed.

note also that I chose not to add a radix or base support for the helper.  Debatalbe choice.